protocols/imap/src/test/java/org/apache/james/imap/processor/base/SelectedMailboxImplTest.java

```diff
@@ -0,0 +1,172 @@
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.MessageMetaData;
+import org.apache.james.mailbox.model.SearchQuery;
+import org.apache.james.mailbox.store.SimpleMessageMetaData;
+import org.apache.james.mailbox.store.event.EventFactory;
+import org.apache.james.mailbox.store.mail.model.DefaultMessageId;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
+
+
+public class SelectedMailboxImplTest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SelectedMailboxImplTest.class);
+    public static final MessageUid MESSAGE_UID_5 = MessageUid.of(5);
+
+    private ExecutorService executorService;
+    private MailboxManager mailboxManager;
+    private MessageManager messageManager;
+    private MailboxPath mailboxPath;
+    private ImapSession imapSession;
+    private Mailbox mailbox;
+
+    @Before
+    public void setUp() throws Exception {
+        executorService = Executors.newFixedThreadPool(1);
+
+        mailboxPath = new MailboxPath(MailboxConstants.USER_NAMESPACE, "tellier@linagora.com", MailboxConstants.INBOX);
+        mailboxManager = mock(MailboxManager.class);
+        messageManager = mock(MessageManager.class);
+        when(mailboxManager.getMailbox(eq(mailboxPath), any(MailboxSession.class))).thenReturn(messageManager);
+        when(messageManager.getApplicableFlags(any(MailboxSession.class))).thenReturn(new Flags());
+        when(messageManager.search(any(SearchQuery.class), any(MailboxSession.class)))
+            .then(sleepThenSearchAnswer());
+
+        imapSession = mock(ImapSession.class);
+        when(imapSession.getAttribute(ImapSessionUtils.MAILBOX_SESSION_ATTRIBUTE_SESSION_KEY)).thenReturn(mock(MailboxSession.class));
+
+        mailbox = mock(Mailbox.class);
+        when(mailbox.generateAssociatedPath()).thenReturn(mailboxPath);
+    }
+
+    @After
+    public void tearDown() {
+        executorService.shutdownNow();
+    }
+
+    @Ignore
+    @Test
+    public void concurrentEventShouldNotSkipUidEmmitedDuringInitialization() throws Exception {
+        final AtomicInteger success = new AtomicInteger(0);
+        doAnswer(generateEmitEventAnswer(success))
+            .when(mailboxManager)
+            .addListener(eq(mailboxPath), any(MailboxListener.class), any(MailboxSession.class));
+
+        SelectedMailboxImpl selectedMailbox = new SelectedMailboxImpl(
+            mailboxManager,
+            imapSession,
+            mailboxPath);
+
+        assertThat(selectedMailbox.getLastUid().get()).isEqualTo(MESSAGE_UID_5);
+    }
+
+    @Ignore
+    @Test
+    public void concurrentEventShouldBeSupportedDuringInitialisation() throws Exception {
```

Comment:
"beSupported" is not precise enough IMO, I think you expect it to process the event successfully